test/realtime/tenants/authorization_test.exs

```diff
@@ -0,0 +1,146 @@
+    channel = channel_fixture(tenant)
+
+    create_rls_policy(db_conn, :select_authenticated_role)
+
+    claims = %{sub: random_string(), role: context.role, exp: Joken.current_time() + 1_000}
+    signer = Joken.Signer.create("HS256", "secret")
+
+    jwt = Joken.generate_and_sign!(%{}, claims, signer)
+
+    %{channel: channel, db_conn: db_conn, jwt: jwt, claims: claims}
+  end
+
+  describe "get_authorizations for Plug.Conn" do
+    @tag role: "authenticated", rls: :select_authenticated_role
+    test "authenticated user has read permissions", %{
+      channel: channel,
+      db_conn: db_conn,
+      jwt: jwt,
+      claims: claims,
+      role: role
+    } do
+      params = %{
+        channel_name: channel.name,
+        headers: [{"header-1", "value-1"}],
+        jwt: jwt,
+        claims: claims,
+        role: role
+      }
+
+      {:ok, conn} =
+        Authorization.get_authorizations(Phoenix.ConnTest.build_conn(), db_conn, params)
+
+      assert {:ok, %{read: true}} = conn.assigns.permissions
+    end
+
+    @tag role: "anon", rls: :select_authenticated_role
+    test "anon user has no read permissions", %{
+      channel: channel,
+      db_conn: db_conn,
+      jwt: jwt,
+      claims: claims,
+      role: role
+    } do
+      params = %{
+        channel_name: channel.name,
+        headers: [{"header-1", "value-1"}],
+        jwt: jwt,
+        claims: claims,
+        role: role
+      }
+
+      {:ok, conn} =
+        Authorization.get_authorizations(Phoenix.ConnTest.build_conn(), db_conn, params)
+
+      assert {:ok, %{read: false}} = conn.assigns.permissions
+    end
+  end
+
+  describe "get_authorizations for Phoenix.Socket" do
+    @tag role: "authenticated", rls: :select_authenticated_role
+    test "authenticated user has read permissions", %{
+      channel: channel,
+      db_conn: db_conn,
+      jwt: jwt,
+      claims: claims,
+      role: role
+    } do
+      params = %{
+        channel_name: channel.name,
+        headers: [{"header-1", "value-1"}],
+        jwt: jwt,
+        claims: claims,
+        role: role
+      }
+
+      {:ok, conn} =
+        Authorization.get_authorizations(
+          Phoenix.ChannelTest.socket(RealtimeWeb.UserSocket),
+          db_conn,
+          params
+        )
+
+      assert {:ok, %{read: true}} = conn.assigns.permissions
+    end
+
+    @tag role: "anon", rls: :select_authenticated_role
+    test "anon user has no read permissions", %{
+      channel: channel,
+      db_conn: db_conn,
+      jwt: jwt,
+      claims: claims,
+      role: role
+    } do
+      params = %{
+        channel_name: channel.name,
+        headers: [{"header-1", "value-1"}],
+        jwt: jwt,
+        claims: claims,
+        role: role
+      }
+
+      {:ok, conn} =
+        Authorization.get_authorizations(
+          Phoenix.ChannelTest.socket(RealtimeWeb.UserSocket),
+          db_conn,
+          params
+        )
+
+      assert {:ok, %{read: false}} = conn.assigns.permissions
```

Comment:
I'm interested in a case where user has `read` permissions but no `write` permissions.

Reply:
potentially that will happen at broadcast where users can send information but won't read it (e.g. sending telemetry)

we will bring some of those use cases after we have write permissions ready